contracts/gauges/LiquidityGaugeV5Collateral.vy

```diff
@@ -0,0 +1,886 @@
+        if is_rewards:
+            self._checkpoint_rewards(_addr, total_supply, _claim_rewards, ZERO_ADDRESS)
+
+        total_supply += _value
+        new_balance: uint256 = self.balanceOf[_addr] + _value
+        new_adjusted_balance: uint256 = self.adjustedBalanceOf[_addr] + _value
+        self.balanceOf[_addr] = new_balance
+        self.adjustedBalanceOf[_addr] = new_adjusted_balance
+        self.totalSupply = total_supply
+
+        self._update_liquidity_limit(_addr, new_adjusted_balance, total_supply)
+
+        ERC20(LP_TOKEN).transferFrom(msg.sender, self, _value)
+
+    log Deposit(_addr, _value)
+    log Transfer(ZERO_ADDRESS, _addr, _value)
+
+
+@external
+@nonreentrant('lock')
+def withdraw(_value: uint256, _claim_rewards: bool = False):
+    """
+    @notice Withdraw `_value` LP tokens
+    @dev Withdrawing also claims pending reward tokens
+    @param _value Number of tokens to withdraw
+    """
+    self._checkpoint(msg.sender)
+
+    if _value != 0:
+        is_rewards: bool = self.reward_count != 0
+        total_supply: uint256 = self.totalSupply
+        if is_rewards:
+            self._checkpoint_rewards(msg.sender, total_supply, _claim_rewards, ZERO_ADDRESS)
+
+        total_supply -= _value
+        new_balance: uint256 = self.balanceOf[msg.sender] - _value
+        new_adjusted_balance: uint256 = self.adjustedBalanceOf[msg.sender] - _value
+        self.balanceOf[msg.sender] = new_balance
+        self.adjustedBalanceOf[msg.sender] = new_adjusted_balance
+        self.totalSupply = total_supply
+
+        self._update_liquidity_limit(msg.sender, new_adjusted_balance, total_supply)
+
+        ERC20(LP_TOKEN).transfer(msg.sender, _value)
+
+    log Withdraw(msg.sender, _value)
+    log Transfer(msg.sender, ZERO_ADDRESS, _value)
+
+
+@internal
+def _transfer(_from: address, _to: address, _value: uint256):
+    self._checkpoint(_from)
+    self._checkpoint(_to)
+
+    if _value != 0:
+        total_supply: uint256 = self.totalSupply
+        is_rewards: bool = self.reward_count != 0
+        is_whitelisted_custody_from: bool = self.whitelisted_custody[_from]
+        is_whitelisted_custody_to: bool = self.whitelisted_custody[_to]
+
+        if is_rewards:
+            self._checkpoint_rewards(_from, total_supply, False, ZERO_ADDRESS)
+        new_balance: uint256 = self.balanceOf[_from] - _value
+        self.balanceOf[_from] = new_balance
+        if(!is_whitelisted_custody_from and !is_whitelisted_custody_to):
+          new_adjusted_balance: uint256 = self.adjustedBalanceOf[_from] - _value
+          self.adjustedBalanceOf[_from] = new_adjusted_balance
+          self._update_liquidity_limit(_from, new_adjusted_balance, total_supply)
+
+        if is_rewards:
+            self._checkpoint_rewards(_to, total_supply, False, ZERO_ADDRESS)
+        new_balance = self.balanceOf[_to] + _value
+        self.balanceOf[_to] = new_balance
+        if(!is_whitelisted_custody_to):
```

Comment:
Line 532 and this line will both be triggered unless we explicitly check `is_whitelisted_custody_from == true`